test/tools/d_do_test.d

```diff
@@ -489,6 +524,38 @@ bool gatherTestParameters(ref TestArgs testArgs, string input_dir, string input_
     if (findTestParameter(envData, file, "POST_SCRIPT", testArgs.postScript))
         testArgs.postScript = replace(testArgs.postScript, "/", to!string(envData.sep));
 
+    string cmds;
+    if (findTestParameter(envData, file, "EXTRA_ARTIFACT", cmds, "\n"))
```

Comment:
```
if (!findTestParameter(envData, file, "EXTRA_ARTIFACT", cmds, "\n"))
    return true;
...
```

Reply:
I prefer the current form to an early exit because it emphazises that all code belongs to `EXTRA_ARTIFACT`.

Reply:
Fair enough